src/nbl/ext/MitsubaLoader/ParserUtil.cpp

```diff
@@ -133,16 +133,17 @@ void ParserManager::parseElement(const Context& ctx, const char* _el, const char
 	
 	if (core::strcmpi(_el, "include") == 0)
 	{
-		auto file = m_filesystem->createAndOpenFile(ctx.currentXMLDir+_atts[1]);
-		if (!file) // try global path
-			file = m_filesystem->createAndOpenFile(_atts[1]);
-		if (!file)
+		system::ISystem::future_t<core::smart_refctd_ptr<system::IFile>> future;
+		bool validInput = m_system->createFile(future, ctx.currentXMLDir.string()+_atts[1], system::IFile::ECF_READ);
+		if (!validInput) // try global path
+			validInput = m_system->createFile(future, _atts[1], system::IFile::ECF_READ);
```

Comment:
I'd get the system pointer from the asset loader override
https://github.com/Devsh-Graphics-Programming/Nabla/blob/androidExPort/include/nbl/asset/interchange/IAssetLoader.h#L183
(make a getter for it)

which is supposed to be in your loaders loading context (c.f. MTL and OBJ loaders).